### ISSUE TITLE
Debug assertion in test case 'qstringlist'

### ISSUE THREAD

**Issue by ThomasKrenn** [copperspice/copperspice#177]
TEST_CASE("QStringList skip_empty_space", "[qstringlist]")

STL Error  "vector iterators incompatible" in 
(QString8.cpp) QString8::cs_internal_find_fast
(MSVC / debug)

pattern_iter is a iterator of strFolded
str.cend() is a iterator of str

```cpp
     auto iter = iter_begin;
     QString8 strFolded = str.toCaseFolded();
     
     while (iter != iter_end)   {
        // review: account for code points which expand when folded
     
        if (iter->toCaseFolded() == QString8(strFolded[0]))  {
           auto text_iter    = iter + 1;
           auto pattern_iter = strFolded.begin() + 1;
     
           while (text_iter != iter_end && pattern_iter != str.cend())  {

```

**Comment by agserm** [comment]
Thank you for reporting this. This looks like something we need to escalate and address rather quickly.

**Comment by agserm** [comment]
This has been resolved in 1c85ed6. Thank you for your report.
